scripts/changelog.sh

```diff
@@ -0,0 +1,255 @@
+#!/bin/bash
+#
+# changelog.sh - Generate a list of new commits between two releases
+# 
+# SPDX-License-Identifier: MPL-2.0
+#
+# (C) 2021 BISDN GmbH
+
+set -e
+
+ONELINE_FORMAT_COMMIT='%C(auto)%h %s'
+ONELINE_FORMAT='%C(auto)%s'
+
+while getopts 'cehp:g:vw:' c
+do
+	case "$c" in
+	c)
+		PRINT_COMMITS=1
+		;;
+	e)
+		PRINT_EMPTY=1
+		;;
+	g)
+		SINGLE_REPO=1
+		GIT_REPO=$OPTARG
+		;;
+	h)
+		PRINT_HELP=1
+		;;
+	p)
+		PREFIX=$OPTARG
+		;;
+	v)
+		PRINT_VERSIONS=1
+		;;
+	w)
+		WORKDIR=$OPTARG
+		;;
+	esac
+done
+
+shift $((OPTIND-1))
+OLD=$1
+NEW=$2
+
+# $1 list of packages in the image
+# $2 packages available and their versions
+collect_package_versions() {
+	local package_list pkg_name pkg_version depends_file
+	declare -n packages=$1
+	declare -n versions=$2
+
+	sed -i 's|^TMPDIR = .*|TMPDIR = "${TOPDIR}/tmp"|' conf/local.conf.sample
+	sed -i 's|^DL_DIR ?= .*|DL_DIR = "${TOPDIR}/dl"|' conf/local.conf.sample
+	sed -i 's|^SSTATE_DIR ?= .*|SSTATE_DIR = "${TOPDIR}/sstate-cache"|' conf/local.conf.sample
+	source ../oe-init-build-env . > /dev/null
+	git checkout conf/local.conf.sample
+	bitbake -g full >/dev/null
+
+	# This is a list of recipe names, not (sub-)packages, but unless we
+	# also want to build an image, this is the closest we can get. 
+	packages=$(grep -v -e '-native' pn-buildlist)
+	oIFS=$IFS
+	IFS="
+"
+	package_list=$(bitbake -s full)
+	for package in $package_list; do
+		pkg_name=$(echo $package | awk '{ print $1}')
+		pkg_version=$(echo $package | awk -F: '{print $2}')
+		versions[$pkg_name]=$pkg_version
+	done
+	IFS=$oIFS
+}
+
+[ -n "$PRINT_HELP" -o -z "$OLD" -o -z "$NEW" ] && {
+	echo "Generate a full release changelog:"
+	echo "Usage: $0 [<options>] <oldversion> <newversion>"
+	echo "Generate a changelog for two branches in a single git repository"
+	echo "Usage: $0 [<options>] -g <uri> <oldversion> <newversion>"
```

Comment:
I would add a working example for both usage types.

Something like:
```
./scripts/changelog.sh v4.3.0 master                                  
./scripts/changelog.sh -g git@github.com:bisdn/basebox.git 1.9.16 1.9.17
```

Reply:
Maybe not use the git transport, if github wants to turn it off ;P

But sure, I can add some examples and more explanation.